CMakeLists.txt

```diff
@@ -815,6 +814,13 @@ if(opencoarrays_aware_compiler)
       add_caf_test(alloc_comp_multidim_shape 2 alloc_comp_multidim_shape)
       set_tests_properties(alloc_comp_multidim_shape PROPERTIES TIMEOUT 300)
     endif()
+    if(NOT CMAKE_Fortran_COMPILER_VERSION VERSION_LESS 15)
+        add_caf_test(team_number 8 team_number)
+        add_caf_test(teams_this_image 8 teams_this_image)
+        add_caf_test(teams_num_images 8 teams_num_images)
+    else()
+        add_caf_test(team_number_pre15 8 team_number_pre15)
```

Comment:
The tests for `team_number` are essentially the same, just with an additional assert that was commented out included. I do not like the idea of maintaining two separate files with practically the exact same content, nor do I like having to include this `else` clause in the CMake file. @vehre would you be able to just give the test file a capital `.F90` file extension and use a preprocessor macro to conditionally include the assertion in the test?

Reply:
Did that. Sorry, for messing that up. On gfortran the policy is to create a new test. So I did it here, too. I've now added the preprocessor directives.